Sources/LiveKit/Core/Room+EngineDelegate.swift

```diff
@@ -194,4 +194,35 @@ extension Room {
             }
         }
     }
+
+    func room(didReceiveTranscriptionPacket packet: Livekit_Transcription) {
+        guard let participant = _state.read({
+            $0.remoteParticipants.values.first { $0.identity == Participant.Identity(from: packet.transcribedParticipantIdentity) }
+        }) else {
+            log("[Transcription] Could not find participant: \(packet.transcribedParticipantIdentity)", .warning)
```

Comment:
@hiroshihorie you'll need to include support for transcription events coming down on the LocalParticipant as well, to record the results of STT in the agent

Reply:
I did a quick spike and it works to just change the delegate methods to receive generic Participant and TrackPublication instead of the Remote variants, change these lines up here to the following

```
        let participant: Participant?
        
        let identity = Participant.Identity(from: packet.transcribedParticipantIdentity)
        if localParticipant.identity == identity {
            participant = localParticipant
        } else {
            participant = _state.read({ $0.remoteParticipants.values.first { $0.identity == identity } })
        }
            
        guard let participant else {
            log("[Transcription] Could not find participant: \(packet.transcribedParticipantIdentity)", .warning)
            return
        }
 ```

Reply:
Thank you ! will update.